code/__HELPERS/_lists.dm

```diff
@@ -809,13 +809,16 @@
 ///return first thing in L which has var/varname == value
 ///this is typecaste as list/L, but you could actually feed it an atom instead.
 ///completely safe to use
-/proc/get_element_by_var(list/inserted_list, varname, value)
+/proc/get_element_by_var(list/inserted_list, varname, value, poll_value = FALSE)
 	varname = "[varname]"
 	for(var/datum/checked_datum in inserted_list)
 		if(!checked_datum.vars.Find(varname))
 			continue
 		if(checked_datum.vars[varname] == value)
-			return checked_datum
+			if(poll_value)
+				return list(checked_datum, checked_datum.vars[varname])
+			else
+				return checked_datum
```

Comment:
this proc not returning a defined type is kinda an issue if it's meant to be a helper and not just a debug tool. what is it for in your mind?
also please document what poll_value actually does this doesn't really make much sense to me

Reply:
I want to be able to check in arbitrary variable on an arbitrarily typed object for an arbitrary value as part of the callback logic returning true or false; I'm not using it in this instance but I want it to be an option

Reply:
Come to think of it however I should still do some compile time checking because even if this doesn't run time it may be improperly used and we'll just be running loops for no reason

Reply:
> Come to think of it however I should still do some compile time checking because even if this doesn't run time it may be improperly used and we'll just be running loops for no reason

And honestly it's outside the scope of fixing this type picky thing